src/main/java/past/cwdt/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -30,9 +35,37 @@ public BookDto getBookById(Long id) {
     }
 
     @Override
-    public List<BookDto> findAll() {
-        return bookRepository.findAll().stream()
+    public List<BookDto> findAll(Pageable pageable) {
+        return bookRepository.findAll(pageable).stream()
                 .map(bookMapper::toDto)
                 .toList();
     }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        checkIfBookExistsById(id);
```

Comment:
Not good practice.
Create method updateBook in your Mapper class
void updateBook(CreateBookRequestDto book, @MappingTarget Book book);

Find book by id
Call updateBook
return bookMapper.toDto(bookRepository.save(book));